### PR TITLE
Removes secondaryParam and hasMore

### PR DESCRIPTION
**Pull request by spacether**
This PR removes the properties
- secondaryParam
- hasMore

because these parameters are booleans that are used to control rendering in mustache templates
We can remove these because mustache already has #-first and #-last and ^-first and ^-last tags
- #secondaryParam -> ^-first
- ^secondaryParam -> -first
- #hasMore -> ^last
- ^hasMore -> -last

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [ ] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/Groovy/api.mustache [hunk 1]
```diff
@@ -49,38 +49,38 @@ class {{classname}} {
...
-        {{#secondaryParam}}
+        {{/-last}}
+        {{#-last}}
+        bodyParams = {{paramName}}
+        {{/-last}}
+        {{/-first}}
+        {{^-first}}
         bodyParams.put("{{baseName}}", {{paramName}})
-        {{/secondaryParam}}
+        {{/-first}}
```
Comment (spacether) [inline modules/openapi-generator/src/main/resources/Groovy/api.mustache:64]: This is so fancy the end result is:
if we have only one item in bodyParams use:
`bodyParams = ParamA`
If we have multiple bodyParams we get:
```
bodyParams. = [:]  # empty map in Groovy
bodyParams.put("ParamA", ParamA
bodyParams.put("ParamB", ParamB
```

**Comment by wing328** [comment]
Looks like the tests need to be updated accordingly as well:
```
[ERROR] COMPILATION ERROR : 
[ERROR] /home/travis/build/OpenAPITools/openapi-generator/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java:[2590,14] cannot find symbol
  symbol:   variable hasMore
  location: variable propA of type org.openapitools.codegen.CodegenProperty
[ERROR] /home/travis/build/OpenAPITools/openapi-generator/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java:[2593,14] cannot find symbol
  symbol:   variable hasMore
  location: variable propB of type org.openapitools.codegen.CodegenProperty
[ERROR] /home/travis/build/OpenAPITools/openapi-generator/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java:[2596,14] cannot find symbol
  symbol:   variable hasMore
  location: variable propC of type org.openapitools.codegen.CodegenProperty
[ERROR] /home/travis/build/OpenAPITools/openapi-generator/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java:[2599,22] cannot find symbol
  symbol:   variable hasMore
  location: variable propBRequired of type org.openapitools.codegen.CodegenProperty
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project openapi-generator: Compilation failure: Compilation failure: 
[ERROR] /home/travis/build/OpenAPITools/openapi-generator/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java:[2590,14] cannot find symbol
[ERROR]   symbol:   variable hasMore
[ERROR]   location: variable propA of type org.openapitools.codegen.CodegenProperty
[ERROR] /home/travis/build/OpenAPITools/openapi-generator/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java:[2593,14] cannot find symbol
[ERROR]   symbol:   variable hasMore
[ERROR]   location: variable propB of type org.openapitools.codegen.CodegenProperty
[ERROR] /home/travis/build/OpenAPITools/openapi-generator/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java:[2596,14] cannot find symbol
[ERROR]   symbol:   variable hasMore
[ERROR]   location: variable propC of type org.openapitools.codegen.CodegenProperty
[ERROR] /home/travis/build/OpenAPITools/openapi-generator/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java:[2599,22] cannot find symbol
[ERROR]   symbol:   variable hasMore
[ERROR]   location: variable propBRequired of type org.openapitools.codegen.CodegenProperty
[ERROR] -> [Help 1]
```

File: modules/openapi-generator/src/main/resources/Groovy/api.mustache (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/Groovy/api.mustache:64]: I think other client generators do not support multiple body parameters so as long as the single body parameter use case works fine, I think that should be good enough.

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team  as well

File: modules/openapi-generator/src/main/resources/Groovy/api.mustache (hunk 1 above)
Comment (spacether) [reply modules/openapi-generator/src/main/resources/Groovy/api.mustache]: Good point. I updated the template to only use bodyParam.

**Comment by spacether** [comment]
> Looks like some tests need to be updated accordingly as well:
@wing328 thank you; all tests are now passing

**Comment by wing328** [comment]
FYI. Some comments still are still referencing `hasMore` in [OneOfImplementorAdditionalData.java](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/OneOfImplementorAdditionalData.java) and we can revise those later as these comments do not impact the functionality of the code.

**Review by wing328 — APPROVED** [review]
LGTM

**Pull request merged**